pkg/apiserver/webhooks.go

```diff
@@ -82,9 +83,48 @@ func NewWebhookConfig() *WebhookConfig {
 		!hasEnv("WEBHOOK_HOST") {
 
 		cfg.Type = WebhookTypeService
-		cfg.ServiceName = getEnv("WEBHOOK_SERVICE_NAME", "api")
-		cfg.ServiceNamespace = getEnv("WEBHOOK_SERVICE_NAMESPACE", "porch-system")
-		cfg.ServiceNamespace = getEnv("CERT_NAMESPACE", cfg.ServiceNamespace)
+
+		var apiSvcNs string
+		cfg.ServiceName = os.Getenv("WEBHOOK_SERVICE_NAME")
+		if cfg.ServiceName == "" { // empty value and unset envvar are the same for our purposes
+			// if WEBHOOK_SERVICE_NAME is not set, try to use the porch API service name
+			apiSvc, err := util.GetPorchApiServiceKey(ctx)
+			if err != nil {
+				panic(fmt.Sprintf("WEBHOOK_SERVICE_NAME environment variable is not set, and could not find porch APIservice: %v", err))
+			}
+			cfg.ServiceName = apiSvc.Name
+			apiSvcNs = apiSvc.Namespace // cache the namespace value to avoid duplicate calls of GetPorchApiServiceKey()
+		}
+		// the webhook service namespace gets it value from the following sources in order of precedence:
```

Comment:
Should this go in the docs somewhere instead?

Reply:
yes. I created an issue: https://github.com/nephio-project/nephio/issues/804